control-plane/api-gateway/cache/consul.go

```diff
@@ -434,9 +434,12 @@ func (c *Cache) Delete(ctx context.Context, ref api.ResourceReference) error {
 		return err
 	}
 
-	options := &api.WriteOptions{}
+	options := &api.WriteOptions{Namespace: ref.Namespace, Partition: ref.Partition}
```

Comment:
Has the namespace here already been mapped (in the situation where the user doesn't have enterprise and/or is using something like `consulDestinationNamespace` [here](https://github.com/hashicorp/consul-k8s/blob/856ade70c85edd235c753d59b83516b2ef842cd0/charts/consul/values.yaml#L1970))?

Reply:
I'm assuming part of the question here also has to do with namespace/partition normalization (i.e. `""` --> `"default"` and CE throwing an error due to a specified namespace)? Because yeah, unless this is coming straight off of an existing ref that we know works with CE, I'd want to vet that this doesn't blow up there.

Reply:
Looks like that happens when the non-normalized key that we use [here](https://github.com/hashicorp/consul-k8s/blob/b9318cb29a2ef8c3d6a8c3170ecb8c20a769e3b8/control-plane/api-gateway/binding/binder.go#L281) is written [here](https://github.com/hashicorp/consul-k8s/blob/b9318cb29a2ef8c3d6a8c3170ecb8c20a769e3b8/control-plane/api-gateway/binding/binder.go#L90)

Reply:
tested this locally using the steps outlined in the description and just removing the ent stuff from the values file, all works as intended